e2etest/l2tests/interface_selector.go

```diff
@@ -173,5 +180,303 @@ var _ = ginkgo.Describe("L2-interface selector", func() {
 				}, 10*time.Second, 1*time.Second).Should(gomega.Not(gomega.HaveOccurred()))
 			}
 		})
+
+		ginkgo.It("Specify not existed interfaces", func() {
+			nonExistedInterface := "ens10"
+			for i := range NodeNics {
+				if NodeNics[i] == nonExistedInterface {
+					framework.Fail("Specify not existed interfaces error, Cluster nodes have the interface named ens10")
+				}
+			}
+			resources := internalconfig.ClusterResources{
+				L2Advs: []metallbv1beta1.L2Advertisement{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "with-interfaces",
+						},
+						Spec: metallbv1beta1.L2AdvertisementSpec{
+							Interfaces: []string{nonExistedInterface},
+						},
+					},
+				},
+			}
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			svc, _ := service.CreateWithBackend(cs, f.Namespace.Name, "lb-service")
+			defer func() {
+				err := cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
+				framework.ExpectNoError(err)
+			}()
+
+			ingressIP := e2eservice.GetIngressPoint(&svc.Status.LoadBalancer.Ingress[0])
+			err = mac.FlushIPNeigh(ingressIP, executor.Host)
+			framework.ExpectNoError(err)
+
+			gomega.Eventually(func() error {
+				return mac.RequestAddressResolutionFromIface(ingressIP, nonExistedInterface, executor.Host)
```

Comment:
nonExistedInterface is not local interfaces
you should foreach local nics

Reply:
ok